### DISCUSSION TITLE
HELP COMPILING: Original Guide Doesnt Work

### DISCUSSION THREAD

**Discussion by hernandito** [filebrowser/filebrowser#1728] [Q&A]
HI Team,

I have been trying and trying to make my own edits to the source files, but I am not having any luck.

- I made one minor addition to one of the .vue files in (/frontend/src/Components)
- Created an Ubuntu Server VM in my server. Did all apt update and apt upgrade.
- I SSH'd into VM and mapped the folder in my server where I unzipped the source.
- I downloaded the Github zip file into my home folder, and extracted.
- I made one minor addition to one of the .vue files in (/frontend/src/Components)
- In the VM I installed NodeJS, NPM, and Midnight Commander via "_sudo apt install npm_"
- Installed GO manually per some web site's guide
- CD'd to the "_frontend_" folder
- Ran "**_npm install_** "(see results in screencap below)
- I got a few warnings and deprecated warnings. I then ran "**_npm audit fix_**". Some warning again came up.
- Ran **_npm install_** (see results in screencap below)
- Ran "**_go mod download_**" - no message returned.
- Ran "**_go build_**" no message returned

At the end, I went looking for a file called **filebrowser** everywhere and could not find anything. In ~/filebrowser/frontend/dist, I ended up with the files I show below. Something tells me there should be a step to further process these files.

I also tried the exact same process above inside a Filebrowser docker... with the same results.

What am I missing? Any help is greatly appreciated.

Thank you,

Screencaptures of results:

**npm install**
![image](https://user-images.githubusercontent.com/787317/146817676-27a71703-d791-47ad-9373-838ca834d996.png)

**npm install fix**
![image](https://user-images.githubusercontent.com/787317/146817892-32a2d310-8c4b-4e19-bb86-29222e021c73.png)

**npm run build**
![image](https://user-images.githubusercontent.com/787317/146819551-1decedb0-584a-47a6-b174-5da7050d196c.png)

**Contents of ~/filebrowser/frontend/dist/**
![image](https://user-images.githubusercontent.com/787317/146820731-d99896bc-7c64-457c-a75a-b129b94e8b9d.png)
![image](https://user-images.githubusercontent.com/787317/146820754-3c8b0691-2250-4501-afc2-17fd9fc198b0.png)

**Comment by ramiresviana** [comment]
Go commands are related to the backend and should be run on the root folder of the project. Running `npm audit fix` is not necessary and warnings should be ignored.

**Reply by hernandito** [reply]
Thank you @ramiresviana. Per your advise above, I was able to generate the Filebrowser executable!! Progress... 

But, when I try to run the docker, I get:

`standard_init_linux.go:228: exec user process caused: no such file or directory`

Googling around, it wants you to correct the Dockerfile to make line ends to be Unix format. I changes this and same results. Here are the contents of my Dockerfile.

          `FROM alpine:latest
          RUN apk --update add ca-certificates \
                               mailcap \
                               curl \
					           bash \
					           mc
          
          HEALTHCHECK --start-period=2s --interval=5s --timeout=3s \
            CMD curl -f http://localhost/health || exit 1
          
          VOLUME /srv
          VOLUME /db
          VOLUME /app
          
          EXPOSE 80
          RUN mkdir -p -m 777 /app
          RUN mkdir -p -m 777 /dv
          
          
          
          COPY .docker.json /.filebrowser.json
          COPY filebrowser /filebrowser
          COPY custom.css /app/custom.css
          COPY database.db /db/database.db
          
          RUN chmod +x /filebrowser
          
          ENTRYPOINT [ "/filebrowser" ]
          CMD ["--root=/data", "--address=0.0.0.0", "--database=/db/database.db", "--noauth" ]`

I have no issues running the docker with the original "filebrowser" file... which leads me to believe it's something in the go backend build.

Any ideas how to overcome this? Is Ubuntu the right environment to build this? 

Thanks again!

**Reply by ramiresviana** [reply]
> > standard_init_linux.go:228: exec user process caused: no such file or directory
> 
> That error is related to an issue with the Go linking, not about the file being nonexistent. To make FileBrowser work on Alpine Linux, CGO needs to be disabled. You can control this by setting the CGO_ENABLED environment variable when running the go tool: set it to 1 to enable the use of cgo, and to 0 to disable it.

https://github.com/filebrowser/filebrowser/issues/1513#issuecomment-906659233

**Reply by hernandito** [reply]
Thank you... I tried Googling... 

Where exactly do I set the CGO_ENABLED=0 variable? Is this in the Ubuntu VM where I am compiling? I looked in the Filebrowser folder and there is a file _**.goreleaser.yml**_ that already has a line CGO_ENABLED=0.  

I tried to add the variable when creating the docker in my server... without luck.

This is my first time trying anything like this.

**Reply by hernandito** [reply]
After some more Googling, I found something that worked... I added this to my Dockerfile:

`RUN apk add --no-cache libc6-compat`

@ramiresviana Thank you! On my way to making some customizations...
